app/models/manageiq/providers/ibm_cloud/inventory/collector/object_storage.rb

```diff
@@ -0,0 +1,29 @@
+class ManageIQ::Providers::IbmCloud::Inventory::Collector::ObjectStorage < ManageIQ::Providers::IbmCloud::Inventory::Collector
+  require_nested :StorageManager
+
+  def buckets
+    buckets = []
+
+    # XXX: filtering out buckets not from our region, as of now the REST call fails with an
+    # exception if used with from different region than the bucket's
+    (connection.list_buckets[:buckets] || []).each do |bucket|
+      connection.get_bucket_location(:bucket => bucket[:name]).location_constraint
```

Comment:
Is there a way to check this that doesn't involve raising an exception?  A nicer way to do this might be

`connection.list_buckets[:buckets].to_a.reject { |bucket| bucket.in_my_region? # this part totally made up but you get the idea }`

Reply:
I'm not sure we can find out which buckets to filter out without using an "exception" handling block. @jaywcarman Who can we ask about this issue on our side?